### PR TITLE
Monitoring conditions: Resolve boolean type coercion bug

### PR DESCRIPTION
**Pull request by devarrowsmith**
Fixes a type coercion bug in the monitoring conditions form, which converted boolean form values from `undefined` to `false` to `true` on subsequent form submissions.

### CODE REVIEW

File: server/controllers/monitoringConditions/monitoringConditionsController.ts [hunk 1]
```diff
@@ -13,8 +13,30 @@ import nextPage, { getSelectedMonitoringTypes } from './nextPage'
 
 const monitoringConditionsFormDataModel = z.object({
   action: z.string().default('continue'),
-  acquisitiveCrime: z.coerce.boolean(),
-  dapol: z.coerce.boolean(),
+  acquisitiveCrime: z.union([z.string(), z.undefined()]).transform(value => {
+    if (value === undefined) {
+      return null
+    }
+    if (value === 'true') {
```
Comment (bjpirt) [inline server/controllers/monitoringConditions/monitoringConditionsController.ts:20]: This can be simplified to `return value === 'true'`

**Review by bjpirt — APPROVED** [review]

File: server/controllers/monitoringConditions/monitoringConditionsController.ts (hunk 1 above)
Comment (bjpirt) [reply server/controllers/monitoringConditions/monitoringConditionsController.ts:20]: o the whole thing could just be `value === undefined ? null : value === 'true'`

Comment (bjpirt) [reply server/controllers/monitoringConditions/monitoringConditionsController.ts:20]: Also, I'm not sure why we'd want to allow a value of anything other than true or false

Comment (devarrowsmith) [reply server/controllers/monitoringConditions/monitoringConditionsController.ts:20]: This is an optional field; the user may not want to provide a value for it. As such, the possible inputs are 'true', 'false' and 'undefined'.

**Review by bjpirt — APPROVED** [review]

**Pull request merged**